### DISCUSSION TITLE
Alerting: Add Equals as a alert condition

### DISCUSSION THREAD

**Discussion by tporeba** [grafana/grafana#68679] [Alerting]
An issue https://github.com/grafana/grafana/issues/19193 has been closed because of moving Grafana feature requests to GitHub Discussions.

> Should this feature still be of interest to you, please re-create it in our Discussions area(https://github.com/grafana/grafana/discussions). 

So here it is, let me quote the original author:

[atlasW](https://github.com/atlasW) wrote:
> I expect the monitored value to trigger an alarm at 0. So I set the alarm condition of "is within range 0 to 0", but he did not trigger the alarm. So I looked at the code and found that'within range'does not include both sides of the boundary. So if I think it's equal to zero, call the police. How should I set it?

I completely agree. 

Currently there are following alert condition evaluators, based on [evaluator.go](https://github.com/grafana/grafana/blob/main/pkg/services/alerting/conditions/evaluator.go):

- IS ABOVE / `gt` -> strict greater than, `x > T`
- IS BELOW / `lt` -> strict lower than, `x < T`
- IS WITHIN RANGE / `within_range` -> strictly between upper and lower thresholds, `x ∈ (T1, T2)`
- IS OUTSIDE RANGE / `outside_range` -> strictly outside upper and lower thresholds, `x ∉ <T1,T2>`
- NO DATA

Why there is at least no "IS EQUAL" condition to complement those?
Is there any other non-obvious way to get equality? 
Is there a way to negate a condition? This would be awkward, but still would allow to construct equality.

I think all those non-strict versions could be useful:
- `x ≥ T`
- `x ≤ T`
- `x ∉ <T1,T2) `
- `x ∉ (T1, T2>`
- `x ∉ (T1, T2)`
- `x ∈ <T1, T2)`
- `x ∈ <T1, T2>`
- `x ∈ (T1, T2>`

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82690. Feel free to continue the discussion around this there. Thank you!
